src/main/scala/io/druid/indexer/spark/SparkDruidIndexer.scala

```diff
@@ -147,9 +146,9 @@ object SparkDruidIndexer extends Logging {
       )
 
     logInfo("Starting uniqes")
-    val optionalDims : Option[Set[String]] = if( dataSchema.getDelegate.getParser.getParseSpec.getDimensionsSpec.hasCustomDimensions) {
+    val optionalDims: Option[Set[String]] = if (dataSchema.getDelegate.getParser.getParseSpec.getDimensionsSpec.hasCustomDimensions) {
       val parseSpec = dataSchema.getDelegate.getParser.getParseSpec
-      Option((Set[String]() ++ parseSpec.getDimensionsSpec.getDimensions) -- parseSpec.getDimensionsSpec.getDimensionExclusions)
+      Option(parseSpec.getDimensionsSpec.getDimensions.asScala.toSet -- parseSpec.getDimensionsSpec.getDimensionExclusions)
```

Comment:
The logic is actually wrong here, when custom dimensions are set, dimension exclusions do not apply according to
https://github.com/druid-io/druid-api/blob/master/src/main/java/io/druid/data/input/impl/MapInputRowParser.java#L48